src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -6344,8 +6344,8 @@ private ExpressionAst HashExpressionRule(Token atCurlyToken, bool parsingSchemaE
                 // Note -  the error handling function inspects the error message body to extra the ParserStrings property name. It uses this value as the errorid.
                 var errorMessageExpression = parsingSchemaElement ?
                     (Expression<Func<string>>)(() => ParserStrings.IncompletePropertyAssignmentBlock)
-                    : (Expression<Func<string>>)(() => ParserStrings.IncompleteHashLiteral);
-                ReportIncompleteInput(keyValuePairs.Any() ? After(keyValuePairs.Last().Item2) : After(atCurlyToken), rCurly.Extent, errorMessageExpression);
+                    : (Expression<Func<string>>)(() => ParserStrings.MissingEndCurlyBrace);
+                ReportIncompleteInput(After(atCurlyToken), rCurly.Extent, errorMessageExpression);
```

Comment:
I see. I tried to fix.

After that there is another question: Are NewLine and Semi terminators having the same rights? Compare:
```powershell

@{
a = 
b =1
c =2
}


@{
a = ;
b =1;
c =2;
}
```
If we will fix this (require that the expression begin immediately after the equal on the same line) of cause it will be a breaking change.